### ISSUE TITLE
Data schema mismatch between merged block

### ISSUE THREAD

**Issue by mapshen** [apache/pinot#7694]
We run a realtime table `table1`  with fields `X` in **upsert** mode. When a new field `Y` is added to the schema, a simple `select` query in the Pinot explorer will return the following error:

```
[
  {
    "message": "MergeResponseError:\nData schema mismatch between merged block:  [X(DOUBLE)] and block to merge:  [X(DOUBLE),Y(DOUBLE)], drop block to merge",
    "errorCode": 500
  }
]
```

The root cause seems to be that not all records are updated to have the `Y` field after the schema change. In the worst case, some record will never receive an update to contain that `Y` field.

**Comment by Jackie-Jiang** [comment]
@mapshen Have you reloaded the table? After adding a field, you need to reload the table in order to generate the new column

**Comment by mapshen** [comment]
@Jackie-Jiang Yes we did. We don't see this error with a regular table. This only manifests when `upsert` is on. Does `reload` rebuild the index `upsert` uses?

**Comment by Jackie-Jiang** [comment]
No for the current consuming segments. We are working on a solution to make reload work on consuming segment with all the available feature support. For now one workaround is to restart the server so that a new consuming segment is created (disable then reenable the table can also serve the purpose)

**Comment by mapshen** [comment]
@Jackie-Jiang it doesn't seem explain why we don't experience such an issue with tables without having `upsert` enabled.

Even in this `table1`, we only receive the error with 

```
select * from table1 limit 10
```

The error will not show up if we do

```
select * from table1 limit 10 option (skipUpsert=True)
```

That said, maybe we are doing the reload incorrectly? Can you let us know the right way to do it?

Also, what is the right way to restart the server so that a consuming segment can be closed and persisted to disk?

**Comment by kkrugler** [comment]
Hi @mapshen - I'd suggest you repost this to the Slack #troubleshooting channel, as that will make it visible to the wider Pinot community, and provides faster/more effective resolution. It also exposes the conversation to other Pinot users, which is often very helpful as they deal with similar problems, thanks.
